### PR TITLE
Record har and screenshot

### PR DESCRIPTION
**Pull request by pes10k**
partially fixes brave/pagegraph-crawl#68

### CODE REVIEW

**Comment by pes10k** [comment]
also update typescript, get the linter and similar working again, address and stop swallowing a bunch of depreciation errors from `argparse`, and only print chromium debugging logs if `--debug verbose`

File: src/run.ts [hunk 1]
```diff
@@ -10,82 +10,95 @@ const defaultShieldsSetting = 'down'
...
-parser.addArgument(['-c','--crawl-duplicates'], {
-  help: 'Enable crawls for redirected URLs that are already present in the redirection chain',
-  action: 'storeTrue',
-  defaultValue: false
+parser.add_argument('-c', '--crawl-duplicates', {
+  help: 'Enable crawls for redirected URLs that are already present in the redirection chain.',
+  action: 'store_true',
+  default: false
+})
+parser.add_argument('--no-screenshot', {
```
[the remark this answers is not in the document]
Comment (pes10k) [reply src/run.ts]: Good point, just made a push to make it off by default.  Once i get HAR's integrated, i'll just add an `--all` option

**Comment by pes10k** [comment]
@ShivanKaul since this PR is getting big, i changed the issue to be only about screenshots, and I'll handle HAR's in a follow up, if that works.  I can start on it after this gets merged

File: src/run.ts [hunk 2]
```diff
@@ -10,82 +10,95 @@ const defaultShieldsSetting = 'down'
...
-  help: 'Enable crawls for redirected URLs that are already present in the redirection chain',
-  action: 'storeTrue',
-  defaultValue: false
+parser.add_argument('-c', '--crawl-duplicates', {
+  help: 'Enable crawls for redirected URLs that are already present in the redirection chain.',
+  action: 'store_true',
+  default: false
+})
+parser.add_argument('--screenshot', {
+  help: 'Do not record a screenshot of the page at the end of the crawl.',
```
Comment (ShivanKaul) [inline src/run.ts:87]:
```suggestion
  help: 'Record a screenshot of the page at the end of the crawl.',
```

**Review by ShivanKaul — APPROVED** [review]
Just a minor comment related to help text of `--screenshot` option.

**Pull request merged**